<?xml version="1.0" ?>
<robot name="pointy">
  
  <link name="base_footprint"/>

  <joint name="joint_x" type="prismatic">
    <parent link="base_footprint"/>
    <child link="link1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1.0" lower="-3" upper="3" velocity=".1"/>
  </joint>

  <link name="link1"/>

  <joint name="joint_y" type="prismatic">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1.0" lower="-3" upper="3" velocity=".1"/>
  </joint>

  <link name="link2"/>

  <joint name="rot_z" type="continuous">
    <parent link="link2"/>
    <child link="eef"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1.0" velocity=".1"/>
  </joint>

  <link name="eef">
     <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.1 0.1 0.1"/>
      </geometry>
      <material name="White"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.1 0.1 0.1"/>
      </geometry>
    </collision>
  </link>
</robot>

